spec.html

```diff
@@ -36502,22 +36502,22 @@ <h1>get RegExp.prototype.flags</h1>
         <emu-alg>
           1. Let _R_ be the *this* value.
           1. If _R_ is not an Object, throw a *TypeError* exception.
-          1. Let _result_ be the empty String.
+          1. Let _codeUnits_ be a new empty List.
           1. Let _hasIndices_ be ToBoolean(? Get(_R_, *"hasIndices"*)).
-          1. If _hasIndices_ is *true*, append the code unit 0x0064 (LATIN SMALL LETTER D) as the last code unit of _result_.
+          1. If _hasIndices_ is *true*, append the code unit 0x0064 (LATIN SMALL LETTER D) to _codeUnits_.
```

Comment:
We can leave the accumulator as a string and do

```suggestion
          1. If _hasIndices_ is *true*, set _result_ to the string-concatenation of _result_ and the code unit 0x0064 (LATIN SMALL LETTER D).
```

Reply:
Yeah, as in #2878. I thought about it, but (when you multiply it by 7), I think it would make the algorithm harder to read.

Alternatively, we could set up a List like
```
«
 «*"hasIndices"*, the code unit 0x0064 (LATIN SMALL LETTER D)»,
 «*"global"*, the code unit 0x0067 (LATIN SMALL LETTER G)»,
 ...
»
```
or
```
«
 «*"d"*, *"hasIndices"*»,
 «*"g"*, *"global"*»,
...
»
```
and then factor out all the repeated code into a loop body. Then the 'appending' code would only appear once and using string-concatenation there would be fine. But looping over a list of tuples-as-lists is not really the spec's style.

Alternative 2:
```
1. Set _result_ to SomeHelperOp(_result_, _R_, *"d"*, *"hasIndices"*).
2. Set _result_ to SomeHelperOp(_result_, _R_, *"g"*, *"global"*).
...
```
But then there's the conceptual/editorial overhead of defining `SomeHelperOp`.

Alternative 3:
```
1. Return the string-concatenation of:
  * SomeHelperOp(_R_, *"d"*, *"hasIndices"*)
  * SomeHelperOp(_R_, *"g"*, *"global"*)
  ...
```
Ditto the overhead of `SomeHelperOp`.

Alternatives 2a + 3a: Define `SomeHelperOp` as an abstract closure, so the reader doesn't have to go to elsewhere to see its definition. And it could capture `_R_`, so you could drop that from the invocations. But that's not how we typically use Abstract Closures.

Reply:
I think building a list of strings, and then joining it in one go, is far cleaner to read than building up a string piece by piece. (an AO for it could be called ListJoin; the records and tuples proposal will likely be adding one anyways eventually)

Reply:
> Yeah, as in #2878. I thought about it, but (when you multiply it by 7), I think it would make the algorithm harder to read.

I don't see how that's the case. It's morally the same as what is there now, just a few more words. And once you comprehend the first one, the pattern is easy to recognise. Building a list and then joining it isn't *bad*, but I don't think it's an improvement over this simpler change.

Reply:
I think building a list then joining it is fine.